### DISCUSSION TITLE
New IKE SA or Rekeyed IKE SA

### DISCUSSION THREAD

**Discussion by Rudug** [strongswan/strongswan#2464] [General]
Hi,
Is there any way to distinguish in Charon plugin callback if the established IKE is newly created one or is created as a result of IKEv2 Rekey.  I am currently listening to ike state change events in HA plugin.  ike_state_change callback with new state as IKE_ESTABLISHED is called prior to ike_rekey, so at that instance I cannot distinguish if new IKE SA is as a result of Rekey. 

Regards,
Vinay Gudur

**Comment by tobiasbrunner** [comment]
I don't think it's possible to distinguish why an SA was established in that callback. You'd have to use the `ike_updown()` and `ike_rekey()` callbacks to do that. Why do you need to know?

**Comment by Rudug** [comment]
Thanks Tobias for your quick reply. ike_updown and ike_rekey callback would not help in this case as ike established is called prior to these calls. This was required as a trigger for event when IKE was established for first time and we dont want to trigger event for IKE rekeys.

**Reply by tobiasbrunner** [reply]
Then use `ike_updown` and ignore `ike_rekey`.
